weekly_data_report.R

```diff
@@ -0,0 +1,131 @@
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(sf))
+suppressPackageStartupMessages(library(knitr))
+
+d <- read_csv(args$file_name,
+              col_types = cols(INTAKE_ID = col_character(),
+                               SCREENING_DECISION = col_character(),
+                               DECISION_DATE = col_character(),
+                               PERSON_ID = col_character(),
+                               RACE = col_character(),
+                               ADDRESS_START = col_date(),
+                               MANDATED_REPORTER = col_character(),
+                               #  REPORTER_PERSON_ID = col_character(),
+                               address_type = col_character(),
+                               address = col_character(),
+                               bad_address = col_logical(),
+                               PO = col_logical(),
+                               lat = col_double(),
+                               lon = col_double(),
+                               score = col_double(),
+                               precision = col_character(),
+                               precise_geocode = col_logical(),
+                               fips_tract_id = col_character(),
+                               fraction_assisted_income = col_double(),
+                               fraction_high_school_edu = col_double(),
+                               median_income = col_double(),
+                               fraction_no_health_ins = col_double(),
+                               fraction_poverty = col_double(),
+                               fraction_vacant_housing = col_double(),
+                               dep_index = col_double()
+              ))
+
+###
+# d_test <- read.csv('test/simulated_jfs_data_geocoded.csv')
+# d <- d_test
+###
+d <- dplyr::mutate(d, DECISION_DATE = dht::check_dates(DECISION_DATE))
+
+# rmarkdown::render(input = '/app/aggregate_data_report.rmd',
+#                   params = list(d = d),
+#                   envir = new.env())
+
+
+tract_to_neighborhood <- readRDS('/app/tract_to_neighborhood.rds')
+neighborhood_shp <- readRDS('/app/ham_neighborhoods_dep_index_shp.rds')
+
+# Overall Summary
+date_min <- min(d$DECISION_DATE)
+date_max <- max(d$DECISION_DATE)
+
+# consider 'SCREENED IN AR' same as 'SCREENED IN'
+for (i in 1:nrow(d)) {
+  if (d$SCREENING_DECISION[i] == 'SCREENED IN AR') {
+    d$SCREENING_DECISION[i] <- 'SCREENED IN'
+  }
+}
+
+d_missing_alleg_add <- d %>%
+  filter(address_type == 'ALLEGATION_ADDRESS',
+         is.na(address)) %>%
+  select(PERSON_ID:address_type)
+
+d_fill_in_address <- d %>%
+  filter(PERSON_ID %in% d_missing_alleg_add$PERSON_ID,
+         address_type == 'CHILD_ADDRESS') %>%
+  group_by(PERSON_ID) %>%
+  arrange(desc(ADDRESS_START)) %>%
+  slice(1) %>%
+  select(INTAKE_ID, address:dep_index)
+
+d_missing_alleg_add <- left_join(d_missing_alleg_add, d_fill_in_address, by = 'PERSON_ID')
+
+d <- d %>%
```

Comment:
Let's make the addresses that are missing their own category for the summary output.  We can use a fake neighborhood name of "Missing".  This will let us know how many were not geocoded to make sure something isn't happening with the data as we automate this in the future.

Reply:
By missing address, do you mean missing from the start from JFS or do you mean not geocoded?

Reply:
let's consider both cases to be missing